### PR TITLE
Add component type register to Engine

### LINKED ISSUE #14: Register available component types within Engine
This is one of the biggest blind spots of the Scent which kinda makes harder to start with it. All component types has to be created and maintained outside. This makes all the interaction with Scent much more annoying and usually means something along these lines:

```
cMyComponent = grabComponentTypeSomewhere('myComponent');
eMyEntity.add(new cMyComponent);
```

What if we could do instead just this ?

```
eMyEntity.add('myComponent');
engine.getNodeType(['myComponent']);
```

To achieve this simplicity, the Engine should be used to store component types and use them whenever the actual name of component is used.

### PR DESCRIPTION
**Pull request by danielkcz**
Fixes #14

### CODE REVIEW

**Comment by danielkcz** [comment]
@martikaljuve I would be grateful if you can have a look at this. Not necessarily to check the code, but if such change makes sense as described. It's something I had nearly completed like half a year ago, but your PR kinda persuaded me to wrap it up and push there also :)

**Comment by martikaljuve** [comment]
Ah nice, I'll check it out later in the evening. I think the idea is good, though magic strings can be annoying to refactor. :)

**Comment by danielkcz** [comment]
Well actually you don't need to use strings since I use ES6 Map there, you can use virtually anything, perhaps Symbol or numbers or wrap the strings into some enum like structure. 

This kinda comes from our need when we don't have components stored globally and we need to specify them before we can actually use them. With a lot of components it becomes hell to maintain that actually.

**Comment by martikaljuve** [comment]
Looks good to me, I would definitely use it. I can link to some typos in the docs if you'd like.

An unrelated question: does it ever make sense to create nodes through `new Scent.Node` as shown in docs/node.md instead of `engine.getNodeType`?

**Comment by danielkcz** [comment]
Please add code comment if you see some typo, that would be very helpful, thanks !

Using `Scent.Node` on it's own could make sense in case you would write your own Engine. I wanted to keep this open so the Engine can be somewhat easily replaced if default one doesn't fit your needs.

File: test/engine.test.coffee [hunk 1]
```diff
@@ -86,6 +138,11 @@ describe 'Engine', ->
             expect(nTest.size).to.equal 2
             expect(nTest.head[ symbols.bEntity ]).to.equal firstEntity
 
+        it 'should apss registered component to node type', ->
```
Comment (martikaljuve) [inline test/engine.test.coffee]: typo: **pa**ss

File: docs/entity.md [hunk 2]
```diff
@@ -79,6 +81,28 @@ For a small performance gain you can also supply target array in first argument
 	entity.getAll(components);
 ```
 
+## Use component provider
+
+Optionally you can pass a function which will be called whenever entity needs to identity component type from whatever you have passed in, eg. string or even Symbol.
```
Comment (martikaljuve) [inline docs/entity.md]: typo: identi**f**y

File: docs/engine.md [hunk 3]
```diff
@@ -69,12 +69,35 @@ Once single action type has run through all registered callback functions, the `
 
 Note that trying to trigger action without single registered handler will basically throw away such actions. This is to prevent stacking up in memory without anyone interested.
 
+## Component registration
+
+Engine contains optional register of component types used to simplify interation by just using component type names instead of grabbing component type reference from somewhere else.
```
Comment (martikaljuve) [inline docs/engine.md]: typo: intera**c**tion

**Pull request merged**